src/utils/utils_part.h

```diff
@@ -61,7 +62,26 @@ static inline void ocf_part_sort(struct ocf_cache *cache)
 	ocf_lst_sort(&cache->lst_part);
 }
 
-static inline ocf_cache_mode_t ocf_part_get_cache_mode(struct ocf_cache *cache,
+static inline bool ocf_part_occ_check_limit(struct ocf_request *req)
+{
+	ocf_cache_t cache = req->cache;
+	uint32_t clines = ocf_engine_unmapped_count(req);
+	uint32_t max = req->cache->user_parts[req->part_id].config->max_size;
+	uint32_t curr_cached = env_atomic_read(
+				&cache->user_parts[req->part_id].runtime->valid_cnt);
+
+	uint32_t test = curr_cached + clines;
+	if (max && test > max) {
+		/* Partition reaches its max occupnacy, some cachelines
+		 * must be evcited first */
+		req->part_evict = true;
```

Comment:
Setting `req->part_evict` in this function is probably not what most of the people would expect. Please move it to engine_common.c to make code flow more obvious.